### PR TITLE
Implement : Bankwise Self-Refresh and Temperature affected leakage simulation feature (New Pull request)

### PR DESCRIPTION
**Pull request by kannoth**
1) Bankwise Self-Refresh.
2) Unit tests for Bankwise Self-Refresh.
3) Update the reference output for all the affected Unit-tests.
4) Change data type the power offset factors Rho and Sigma from int64_t to double.
5) Added include guards for MemPowerSpec and MemTimingSpec which was missing.

### CODE REVIEW

**Comment by kannoth** [comment]
@efzulian and @myzinsky  New pull-request opened for tukl-msd repo.

**Comment by efzulian** [comment]
@kannoth thanks! I'm going to review it soon.

File: memspecs/SAMSUNG_M471B5674QH0_2GB_DDR3-800_64bit_Leakage.xml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      <parameter id="idd5" type="double" value="394.2" />
+      <parameter id="TempCfIdd5Alpha" type="double" value="9.816" />
+      <parameter id="TempCfIdd5Beta" type="double" value="0.03788" />
+      <parameter id="idd6" type="double" value="21.51" />
+      <parameter id="TempCfIdd6Alpha" type="double" value="0.7072" />
+      <parameter id="TempCfIdd6Beta" type="double" value="0.04548" />
+      <parameter id="vdd" type="double" value="1.5" />
+  </mempowerspec>
+  <membankwiseparams>
+    <parameter id="rho" type="uint" value ="100" />
```
Comment (myzinsky) [inline memspecs/SAMSUNG_M471B5674QH0_2GB_DDR3-800_64bit_Leakage.xml]: why don't you have here the real measured sigma and rho values?

File: memspecs/SAMSUNG_M471B5674QH0_2GB_DDR3-800_64bit_Leakage.xml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      <parameter id="idd4w" type="double" value="400.8" />
+      <parameter id="TempCfIdd4wAlpha" type="double" value="10.71" />
+      <parameter id="TempCfIdd4wBeta" type="double" value="0.0289" />
+      <parameter id="idd4r" type="double" value="408.8" />
+      <parameter id="TempCfIdd4rAlpha" type="double" value="10.67" />
+      <parameter id="TempCfIdd4rBeta" type="double" value="0.02925" />
+      <parameter id="idd5" type="double" value="394.2" />
+      <parameter id="TempCfIdd5Alpha" type="double" value="9.816" />
+      <parameter id="TempCfIdd5Beta" type="double" value="0.03788" />
+      <parameter id="idd6" type="double" value="21.51" />
```
Comment (myzinsky) [inline memspecs/SAMSUNG_M471B5674QH0_2GB_DDR3-800_64bit_Leakage.xml]: have we not discussed to have something like this:

```
 <parameter id="idd5" type="double" value="394.2" alpha="9.816" beta="0.03788"/>
```

?

File: memspecs/SAMSUNG_M471B5674QH0_2GB_DDR3-800_64bit_Leakage.xml [hunk 3]
```diff
@@ -0,0 +1,80 @@
+<!DOCTYPE memspec SYSTEM "memspec.dtd">
```
Comment (myzinsky) [inline memspecs/SAMSUNG_M471B5674QH0_2GB_DDR3-800_64bit_Leakage.xml]: The name of the file should be:

memspecs/SAMSUNG_M471B5674QH0_2GB_DDR3-800_64bit_Measured.xml

File: memspecs/JEDEC_256Mb_WIDEIO_SDR-200_128bit.xml [hunk 4]
```diff
@@ -64,4 +64,8 @@
       <parameter id="vdd" type="double" value="1.8" />
       <parameter id="vdd2" type="double" value="1.2" />
   </mempowerspec>
+  <membankwiseparams>
```
Comment (myzinsky) [inline memspecs/JEDEC_256Mb_WIDEIO_SDR-200_128bit.xml]: I think we should not put this to every file, only to the measured ones, what do you think? 
If this section is not present we use the default value.

File: memspecs/SAMSUNG_M471B5674QH0_2GB_DDR3-800_64bit_Leakage.xml (hunk 3 above)
Comment (knox357) [reply memspecs/SAMSUNG_M471B5674QH0_2GB_DDR3-800_64bit_Leakage.xml]: Makes sense. I will change the name.

**Pull request closed without merging**